plugin/sycl/tree/hist_updater.cc

```diff
@@ -501,8 +501,8 @@ void HistUpdater<GradientSumT>::InitData(
     hist_local_worker_.Init(qu_, nbins);
 
     hist_buffer_.Init(qu_, nbins);
-    size_t buffer_size = kBufferSize;
-    hist_buffer_.Reset(kBufferSize);
+    size_t buffer_size = 4 * qu_->get_device().get_info<::sycl::info::device::max_compute_units>();
```

Comment:
Does it worth removing `kBufferSize`  constant from the header?
It looks like it is not used now.

Also it would be great to describe the meaning of the multiplier `4`.

Reply:
I have completely reweighted this logic with the new dispatcher. Now one estimate the required buffer size, that reduce memory consumption (critical for customer-class devices).